src/connector/codec/src/decoder/protobuf/parser.rs

```diff
@@ -180,10 +180,43 @@ pub fn from_protobuf_value<'a>(
             ScalarImpl::List(ListValue::new(builder.finish()))
         }
         Value::Bytes(value) => borrowed!(&**value),
-        _ => {
-            return Err(AccessError::UnsupportedType {
-                ty: format!("{kind:?}"),
-            });
+        Value::Map(map) => {
+            let err = || {
+                AccessError::TypeError {
+                    expected: type_expected.to_string(),
+                    got: format!("{:?}", kind),
+                    value: value.to_string(), // Protobuf TEXT
+                }
+            };
+
+            let DataType::Map(map_type) = type_expected else {
+                return Err(err());
+            };
+            let map_desc = kind.as_message().ok_or_else(err)?;
+            if !map_desc.is_map_entry() {
+                return Err(err());
+            }
+
+            let mut key_builder = map_type.key().create_array_builder(map.len());
+            let mut value_builder = map_type.value().create_array_builder(map.len());
+            // NOTE: HashMap's iter order is non-deterministic, but MapValue's
+            // order matters. We sort by key here to have deterministic order
+            // in tests. We might consider removing this, or make all MapValue sorted
+            // in the future.
+            for (key, value) in map.iter().sorted_by_key(|(k, _v)| *k) {
+                key_builder.append(from_protobuf_value(
+                    field_desc,
+                    &key.clone().into(),
+                    map_type.key(),
+                )?);
+                value_builder.append(from_protobuf_value(field_desc, value, map_type.value())?);
```

Comment:
`map_desc.map_entry_value_field()`

The error was not detected as most branches does not check input parameter `field_desc`, with struct / list being the exceptions. Can we test with a map where its value type is a struct?